PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -1601,7 +1601,7 @@ bool FileManager::loadFileData(Document doc, int64_t fileSize, const TCHAR * fil
 		fileFormat._eolFormat = ndds._format;
 
 		//for empty files, if the default for new files is UTF8, and "Apply to opened ANSI files" is set, apply it
-		if (fileSize == 0)
+		if ((fileSize == 0) && (fileFormat._encoding < 1))

```

Comment:
@xomx 
Thank you for your PR and very detail [explanation](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11824#issue-1280131509). 
As @Yaron10 said, I cannot reproduce the crash after https://github.com/notepad-plus-plus/notepad-plus-plus/commit/d1347d501451e10c5d988282ca681760d2c2abff.

My question is why this specific condition: `(fileFormat._encoding < 1)` ?

Reply:
@xomx 
OK, I see. By putting a messagebox for displaying encoding value, I saw the infinite loop - the condition to filter 1 and greater than 1 works to stop the infinite loop.

@Yaron10 
 https://github.com/notepad-plus-plus/notepad-plus-plus/commit/d1347d501451e10c5d988282ca681760d2c2abff (the optimization of speed) prevent program from crash (infinite loop), but the loop does happen, it just stop it at the end of ~10 times.